contracts/core/libs/attestations-registry/AttestationsRegistryConfigLogic.sol

```diff
@@ -145,4 +311,62 @@ contract AttestationsRegistryConfigLogic is
     _authorizedRanges[issuer].pop();
     emit IssuerUnauthorized(issuer, firstCollectionId, lastCollectionId);
   }
+
+
+  function _registerAttestationsCollectionTag(
+    uint256 collectionId,
+    uint8 tagIndex,
+    uint8 tagPower
+  ) internal {
+    if (_tags[tagIndex] == 0) {
+      revert TagDoesNotExist(tagIndex);
+    }
+
+    uint256 currentTags = _attestationsCollectionTagsBitmap[collectionId];
+
+    _attestationsCollectionTagsBitmap[collectionId] = currentTags._addTag(tagIndex, tagPower);
+
+    emit AttestationsCollectionTagRegistered(collectionId, tagIndex, tagPower);
+  }
+
+  function _unregisterAttestationsCollectionTag(uint256 collectionId, uint8 tagIndex) internal {
+    uint256 currentTags = _attestationsCollectionTagsBitmap[collectionId];
+
+    _attestationsCollectionTagsBitmap[collectionId] = currentTags._removeTag(tagIndex);
+
+    emit AttestationsCollectionTagUnregistered(collectionId, tagIndex);
+  }
+
+
+  function _createNewTag(uint8 tagIndex, bytes32 tagName) internal {
+    if (_tags[tagIndex] != 0) {
+      revert TagAlreadyExists(tagIndex);
+    }
+    tagIndex._checkTagIndexIsValid();
+
+    _tags[tagIndex] = tagName;
+
+    emit NewTagInserted(tagIndex, tagName);
+  }
+
+  function _updateTagName(uint8 tagIndex, bytes32 newtagName) internal {
+    if (_tags[tagIndex] == 0) {
+      revert TagDoesNotExist(tagIndex);
```

Comment:
Not sure if check is needed. May overcomplexify.

If we keep the throws, it should be in the external functions